eZ/Bundle/EzPublishCoreBundle/Command/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,241 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:urls:regenerate-aliases')
+            ->setDescription(
+                'Regenerates Location URL aliases (autogenerated) and cleans up custom Location ' .
+                'and global URL aliases stored in the Legacy Storage Engine'
+            )
+            ->addOption(
+                'iteration-count',
+                'c',
+                InputOption::VALUE_OPTIONAL,
+                'Number of Locations fetched into memory and processed at once',
+                self::DEFAULT_ITERATION_COUNT
+            )
+            ->setHelp(
+                <<<EOT
+<error>Make sure to back up your database before proceeding.</error>
+
+The command <info>%command.name%</info> regenerates URL aliases for Locations and cleans up
+corrupted URL aliases (pointing to non-existent Locations).
+Existing aliases are archived (will redirect to the new ones).
+
+<error>During the script execution the database should not be modified.</error>
+
+Since this script can potentially run for a very long time, to avoid memory exhaustion run it in
+production environment using <info>--env=prod</info> switch and make sure to either disable PHP memory limit
+or use sufficiently big value.
+
+For example, processing of 300 000 Locations can take up to 1 GB of RAM.
+
+Due to performance issues the command does not send any Signals.
+
+EOT
+            );
+    }
+
+    /**
+     * Regenerate URL aliases.
+     *
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $iterationCount = (int)$input->getOption('iteration-count');
+
+        $locationsCount = $this->repository->sudo(
+            function (Repository $repository) {
+                return $repository->getLocationService()->countAllLocations();
+            }
+        );
+
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion(
+            sprintf(
+                '<info>Found %d Locations. Do you want to proceed? [y/N] </info>',
+                $locationsCount
+            ),
+            false
+        );
+        if (!$helper->ask($input, $output, $question)) {
+            return;
+        }
+
+        $output->writeln('Regenerating System URL aliases...');
+
+        $progressBar = $this->getProgressBar($locationsCount, $output);
+        $progressBar->start();
+
+        for ($offset = 0; $offset <= $locationsCount; $offset += $iterationCount) {
+            $locations = $this->repository->sudo(
+                function (Repository $repository) use ($offset, $iterationCount) {
+                    return $repository->getLocationService()->loadAllLocations($offset, $iterationCount);
+                }
+            );
+            $this->processLocations($locations, $progressBar);
+            // force garbage collection
+            gc_collect_cycles();
```

Comment:
Afaik it's fixed in PHP 7.3 but could be you get some extra perf by disabling gc within each iteration, ref:
https://blog.blackfire.io/performance-impact-of-the-php-garbage-collector.html
https://tideways.com/profiler/blog/how-to-optimize-the-php-garbage-collector-usage-to-improve-memory-and-performance

Reply:
> Afaik it's fixed in PHP 7.3 but could be you get some extra perf by disabling gc within each iteration, ref:
> https://blog.blackfire.io/performance-impact-of-the-php-garbage-collector.html
> https://tideways.com/profiler/blog/how-to-optimize-the-php-garbage-collector-usage-to-improve-memory-and-performance

Hmm, so I investigated it further today and I don't see any increase in performance both time and memory-wise. But maybe with PHP 7.3 it will make a difference. Changed via 4a584bc.